src/action.js

```diff
@@ -29,39 +36,35 @@ async function run() {
 
     const token = core.getInput("github-token");
     const octokit = github.getOctokit(token);
-    const data = await octokit.graphql(
-      `
-      query getLinkedIssues($owner: String!, $name: String!, $number: Int!) {
-        repository(owner: $owner, name: $name) {
-          pullRequest(number: $number) {
-            closingIssuesReferences {
-              totalCount
-            }
-          }
-        }
-      }
-      `,
-      {
-        owner: owner.login,
-        name,
-        number,
-      }
-    );
+    const data = await getLinkedIssues(octokit, name, number, owner.login);
 
     core.debug(`
     *** GRAPHQL DATA ***
     ${format(data)}
     `);
 
-    const linkedIssuesCount =
-      data?.repository?.pullRequest?.closingIssuesReferences?.totalCount;
+    const pullRequest = data?.repository?.pullRequest;
+    const linkedIssuesCount = pullRequest?.closingIssuesReferences?.totalCount;
 
     core.setOutput("linked_issues_count", linkedIssuesCount);
 
     if (!linkedIssuesCount) {
-      core.setFailed(
-        `No linked issues found. Please add the corresponding issues in the pull request description.`
-      );
+      const subjectId = pullRequest?.id;
+
+      if (subjectId) {
+        await addComment(octokit, subjectId);
```

Comment:
Apart from that I think we also need to check if the comment exists in any case (to avoid add multiple comments when PR gets updated or failing to delete an unexisting comment), so more or less I'd do:

```js
const shouldComment = core.getInput("comment");
const linkedIssuesCount = await getLinkedIssues(…);
const botComment = shouldComment && await getPrCommentFromGithubBot(); <- this would run the paginated query and filter

if (linkedIssuesCount === 0) {
  core.setFailed('…');
  if (shouldComment && !botComment) {
    // ADD COMMENT
   }
} else {
   core.setOutput("linked_issues_count", linkedIssuesCount);
   if (botComment) {
    // DELETE COMMENT
   }
}
```

Reply:
That's a good idea